frontend/src/Components/FavoriteCard.tsx

```diff
@@ -0,0 +1,71 @@
+import { CombinedFavorite } from '@/common';
+import { StarIcon as SolidStar } from '@heroicons/react/24/solid';
+import { useNavigate } from 'react-router-dom';
+
+interface FavoriteCardProps {
+    favorite: CombinedFavorite;
+    onUnfavorite: () => void;
+}
+
+const FavoriteCard: React.FC<FavoriteCardProps> = ({
+    favorite,
+    onUnfavorite
+}) => {
+    const navigate = useNavigate();
+
+    const handleCardClick = () => {
+        if (favorite.visibility_status) {
+            return;
+        }
+        const state = {
+            subLink: favorite.content_url
+        };
+        if (favorite.type === 'video') {
+            navigate(`/viewer/videos/${favorite.content_id}`);
+        } else if (favorite.type === 'library') {
+            navigate(`/viewer/libraries/${favorite.content_id}`, { state });
+        }
+    };
+
+    return (
+        <div
+            className={`card relative rounded-lg p-3 shadow-md ${
+                favorite.visibility_status ? 'bg-grey-2' : 'bg-inner-background'
+            } ${
+                favorite.visibility_status
+                    ? 'cursor-not-allowed'
+                    : 'cursor-pointer '
+            }`}
+            style={{ width: '220px', height: '220px' }}
+            onClick={favorite.visibility_status ? undefined : handleCardClick}
+        >
+            <div
+                className="absolute top-2 right-2 cursor-pointer"
+                onClick={(e) => {
+                    e.stopPropagation();
+                    onUnfavorite();
+                }}
+            >
+                <SolidStar className="w-5 text-primary-yellow" />
+            </div>
+
+            <figure className=" w-full mx-auto object-contain bg-contain">
+                <img
+                    src={favorite.thumbnail_url}
+                    alt={favorite.name}
+                    className="w-full h-28 object-contain rounded-md mb-3"
+                />
+            </figure>
+            <h3 className="text-lg font-bold text-header-text text-center mb-1 line-clamp-1">
+                {favorite.name}
+            </h3>
```

Comment:
`h3` should already have most necessary classes applied, such as text size, color, and weight. there are a few other things that i changed just to match the cards on the other pages in terms of spacing, etc
```
<div
            className={`card relative p-4 space-y-2 ${
                favorite.visibility_status
                    ? 'bg-grey-2 cursor-not-allowed'
                    : 'bg-inner-background cursor-pointer'
            }`}
            onClick={favorite.visibility_status ? undefined : handleCardClick}
        >
            <div
                className="absolute top-2 right-2 cursor-pointer"
                onClick={(e) => {
                    e.stopPropagation();
                    onUnfavorite();
                }}
            >
                <SolidStar className="w-5 text-primary-yellow" />
            </div>
            <img
                src={favorite.thumbnail_url}
                alt={favorite.name}
                className="h-16 mx-auto object-contain"
            />
            <h3 className="body text-center line-clamp-2">{favorite.name}</h3>
            <p className="body-small text-center">
                {favorite.type === 'video'
                    ? favorite.channel_title
                    : favorite.provider_name}
            </p>
        </div>
```